tests/clv/models/test_pareto_nbd.py

```diff
@@ -331,3 +334,97 @@ def test_expected_purchase_probability(self, test_n, test_t):
             est_purchases_new_customer.mean(("chain", "draw")),
             rtol=0.001,
         )
+
+    def test_dropout_purchase_distributions(self) -> None:
+        # TODO: Why isn't self.data working here?
+        test_data = pd.read_csv("datasets/clv_quickstart.csv")
+        test_data["customer_id"] = test_data.index
+        model = ParetoNBDModel(
+            data=test_data,
+        )
+        model.build_model()
+
+        # TODO: A test with more than 1 chain and draw is required for full coverage.
+        chains = 2
+        draws = 50
+        mock_fit = az.from_dict(
+            {
+                "r": self.rng.normal(self.r_true, 1e-3, size=(chains, draws)),
+                "alpha": self.rng.normal(self.alpha_true, 1e-3, size=(chains, draws)),
+                "s": self.rng.normal(self.s_true, 1e-3, size=(chains, draws)),
+                "beta": self.rng.normal(self.beta_true, 1e-3, size=(chains, draws)),
+            }
+        )
+        rtol = 0.17
+
+        mock_fit = az.from_dict(
+            {
+                "r": [self.r_true],
+                "alpha": [self.alpha_true],
+                "s": [self.s_true],
+                "beta": [self.beta_true],
+            }
+        )
+        model.idata = mock_fit
+
+        rng = np.random.default_rng(42)
+        customer_dropout = model.distribution_new_customer_dropout(random_seed=rng)
+        customer_purchase_rate = model.distribution_new_customer_purchase_rate(
+            random_seed=rng
+        )
+
+        assert isinstance(customer_dropout, xarray.DataArray)
+        assert isinstance(customer_purchase_rate, xarray.DataArray)
+
+        N = 1000
+        lam = pm.Gamma.dist(alpha=self.r_true, beta=1 / self.alpha_true, size=N)
+        mu = pm.Gamma.dist(alpha=self.s_true, beta=1 / self.beta_true, size=N)
+        rtol = 0.11
+
+        np.testing.assert_allclose(
+            customer_purchase_rate.mean(),
+            pm.draw(lam.mean(), random_seed=rng),
+            rtol=rtol,
+        )
+        np.testing.assert_allclose(
+            customer_purchase_rate.var(),
+            pm.draw(lam.var(), random_seed=rng),
+            rtol=rtol,
+        )
+        np.testing.assert_allclose(
+            customer_dropout.mean(), pm.draw(mu.mean(), random_seed=rng), rtol=rtol
+        )
+        np.testing.assert_allclose(
+            customer_dropout.var(), pm.draw(mu.var(), random_seed=rng), rtol=rtol
+        )
+
+    def test_save_load_pareto_nbd(self):
+        # TODO: Why isn't self.data working here?
```

Comment:
Unclear TODO, can we remove it?
```suggestion
        # TODO: Why isn't self.data working here?
```

Reply:
Test data is currently being loaded in the `setup_class` method as a class attribute, but is not being recognized by these tests for some reason. I've made a note to create a `pytest` fixture for the test dataset in a future PR, and modified these TODOs accordingly.